windows-agent/internal/proservices/certs_test.go

```diff
@@ -0,0 +1,47 @@
+package proservices_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	ps "github.com/canonical/ubuntu-pro-for-wsl/windows-agent/internal/proservices"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewTLSCertificates(t *testing.T) {
+	t.Parallel()
+	testcases := map[string]struct {
+		breakDestDir bool
+		breakKeyFile string
+
+		wantErr bool
+	}{
+		"Success": {},
+		"Error when the destination directory cannot be written into": {breakDestDir: true, wantErr: true},
+		"Error when the server private key cannot be written":         {breakKeyFile: "server_key.pem", wantErr: true},
+		"Error when the client private key cannot be written":         {breakKeyFile: "client_key.pem", wantErr: true},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			dir := t.TempDir()
+			if tc.breakDestDir {
+				dir = filepath.Join(dir, "inexistent")
+			}
+
+			if tc.breakKeyFile != "" {
+				err := os.MkdirAll(filepath.Join(dir, tc.breakKeyFile), 0700)
+				require.NoError(t, err, "Setup: could not write directory that should break %s", tc.breakKeyFile)
+			}
+
+			_, err := ps.NewTLSCertificates(dir)
+			if tc.wantErr {
+				require.Error(t, err, "NewTLSCertificates should have failed")
+				return
+			}
+			require.NoError(t, err, "NewTLSCertificates failed")
+		})
+	}
+}
```

Comment:
Just to confirm (it’s ok as it’s not exported, covered by other package tests when calling `New()` and the function is short): you are not testing this one as unit test because you think it’s trivial?

Reply:
Not only because it's trivial, but because we'd be testing more the `x509.CertPool` then our logic, which is a different condition from when testing itt through `proservices.New()`

Reply:
Notice that we can panic but we can't error this function out.

Reply:
agreed!